src/gateway/types.js

```diff
@@ -234,6 +234,11 @@ export type RenderAPI = {
      * Callback to start a trace session for tracing an operation.
      */
     +trace: TraceCallback,
+
+    /**
+     * A logger to use for logging during the render operation.
+     */
+    +logger: Logger,
```

Comment:
While the whole type could be wrapped in `$ReadOnly<{ ... }>`, I can totally understand wanting to use `+` from a documentation point of view.

Reply:
Yeah, I find `$ReadOnly` kinda cumbersome at the type level. I'd really love a `readonly` keyword like C# has.